plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/thrift/TestHiveMetastoreReadAccessOperations.java

```diff
@@ -0,0 +1,340 @@
+                ImmutableMap.of(GET_ALL_VIEWS_WITH_PARAMETER, 1));
+    }
+
+    @Test
+    public void testSelectViewsWithLikeOverSchema()
+    {
+        assertMetastoreInvocations("SELECT * FROM information_schema.views WHERE table_schema LIKE 'test%'",
+                ImmutableMap.of(
+                        GET_ALL_DATABASES, 1,
+                        GET_ALL_VIEWS_WITH_PARAMETER, TEST_TABLES_IN_SCHEMA_COUNT));
+    }
+
+    @Test
+    public void testSelectViewsWithFilterByTableName()
+    {
+        assertMetastoreInvocations("SELECT * FROM information_schema.views WHERE table_name = 'test_table_0'",
+                ImmutableMap.of(
+                        GET_ALL_DATABASES, 1,
+                        GET_ALL_VIEWS_WITH_PARAMETER, TEST_TABLES_IN_SCHEMA_COUNT));
+    }
+
+    @Test
+    public void testSelectViewsWithLikeOverTableName()
+    {
+        assertMetastoreInvocations("SELECT * FROM information_schema.views WHERE table_name LIKE 'test%'",
+                ImmutableMap.of(
+                        GET_ALL_DATABASES, 1,
+                        GET_ALL_VIEWS_WITH_PARAMETER, TEST_TABLES_IN_SCHEMA_COUNT));
+    }
+
+    @Test
+    public void testSelectColumnsWithoutPredicate()
+    {
+        assertMetastoreInvocations("SELECT * FROM information_schema.columns",
+                ImmutableMap.of(
+                        GET_ALL_DATABASES, 1,
+                        GET_ALL_TABLES_WITH_PARAMETER, TEST_TABLES_IN_SCHEMA_COUNT,
+                        GET_ALL_VIEWS_WITH_PARAMETER, TEST_TABLES_IN_SCHEMA_COUNT,
+                        GET_TABLE, TEST_ALL_TABLES_COUNT));
+    }
+
+    @Test
+    public void testSelectColumnsFilterByInformationSchema()
+    {
+        assertMetastoreInvocations("SELECT * FROM information_schema.columns WHERE table_schema = 'information_schema'", ImmutableMap.of());
+    }
+
+    @Test
+    public void testSelectColumnsFilterBySchema()
+    {
+        assertMetastoreInvocations("SELECT * FROM information_schema.columns WHERE table_schema = 'test_schema_0'",
+                ImmutableMap.of(
+                        GET_ALL_TABLES_WITH_PARAMETER, 1,
+                        GET_ALL_VIEWS_WITH_PARAMETER, 1,
+                        GET_TABLE, TEST_TABLES_IN_SCHEMA_COUNT));
+    }
+
+    @Test
+    public void testSelectColumnsWithLikeOverSchema()
+    {
+        assertMetastoreInvocations("SELECT * FROM information_schema.columns WHERE table_schema LIKE 'test%'",
+                ImmutableMap.of(
+                        GET_ALL_DATABASES, 1,
+                        GET_ALL_TABLES_WITH_PARAMETER, TEST_TABLES_IN_SCHEMA_COUNT,
+                        GET_ALL_VIEWS_WITH_PARAMETER, TEST_TABLES_IN_SCHEMA_COUNT,
+                        GET_TABLE, TEST_ALL_TABLES_COUNT));
+    }
+
+    @Flaky(issue = "For particular query there are 10294+/-1 GET_TABLE requests, therefore it is flaky", match = ".*")
```

Comment:
Should I disable that test?

Reply:
where does `+ 91` come from?

i think the best way about this would be to add determinism. perhaps we need some sorting in `calculatePrefixesWithTableName`.
actually, the `getRedirectionAwareTableHandle` done there redundant when redirects are not enabled (which is default)

removing 
```
.filter(objectName -> {
    if (!isColumnsEnumeratingTable(informationSchemaTable) ||
            metadata.isMaterializedView(session, objectName) ||
            metadata.isView(session, objectName)) {
        return true;
    }

    // This is a columns enumerating table and the object is not a view
    try {
        // Table redirection to enumerate columns from target table happens later in
        // MetadataListing#listTableColumns, but also applying it here to avoid incorrect
        // filtering in case the source table does not exist or there is a problem with redirection.
        return metadata.getRedirectionAwareTableHandle(session, objectName).getTableHandle().isPresent();
    }
    catch (TrinoException e) {
        if (e.getErrorCode().equals(TABLE_REDIRECTION_ERROR.toErrorCode())) {
            // Ignore redirection errors for listing, treat as if the table does not exist
            return false;
        }

        throw e;
    }
})
```

makes the test deterministic, and faster (fewer calls)

Reply:
This is preexisting problem. Let's relax the assertion and fix it as separate effort.

Reply:
agreed. maybe

```java
@Test
public void testSelectColumnsFilterByTableName()
{
    metastore.resetCounters();
    computeActual("SELECT * FROM information_schema.columns WHERE table_name = 'test_table_0'");
    Multiset<Methods> invocations = metastore.getMethodInvocations();

    assertThat(invocations.count(GET_TABLE)).as("GET_TABLE invocations")
            // some lengthly explanatory comment why variable count
            // TODO switch to assertMetastoreInvocations when GET_TABLE invocation count becomes deterministic
            // TODO this is horribly lot, why, o why we do those redirect-related calls in ... even if redirects not enabled?????????!!?!1
                    .isBetween(TEST_ALL_TABLES_COUNT + 85, TEST_ALL_TABLES_COUNT + 95);
    invocations = HashMultiset.create(invocations);
    invocations.elementSet().remove(GET_TABLE);

    assertThat(invocations).as("invocations except of GET_TABLE")
            .isEqualTo(ImmutableMultiset.<Methods>builder()
                    .add(GET_ALL_DATABASES)
                    .addCopies(GET_ALL_TABLES_WITH_PARAMETER, TEST_SCHEMAS_COUNT)
                    .addCopies(GET_ALL_VIEWS_WITH_PARAMETER, TEST_SCHEMAS_COUNT)
                    .build());
}

Reply:
@findepi Thanks for the suggestion with awesome comment ;)